### PR TITLE
Moved Pipelines.Testing APIs to Piplelines.Extensions dll

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
cc: @davidfowl, @joshfree, @ahsonkhan

### CODE REVIEW

File: tests/System.Buffers.Experimental.Tests/BufferExtensionTests.cs [hunk 1]
```diff
@@ -1,7 +1,11 @@
-﻿using System.IO.Pipelines.Testing;
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Buffers;
+using System.IO.Pipelines.Testing;
 using Xunit;
 
-namespace System.Buffers.Tests
+namespace System.IO.Pipelines.Tests
```
Comment (ahsonkhan) [inline tests/System.Buffers.Experimental.Tests/BufferExtensionTests.cs]: Should we move this file to tests\System.IO.Pipelines.Tests?

Comment (KrzysztofCwalina) [reply tests/System.Buffers.Experimental.Tests/BufferExtensionTests.cs]: The file has a dependency on System.Buffers.Experimental

Comment (ahsonkhan) [reply tests/System.Buffers.Experimental.Tests/BufferExtensionTests.cs]: I see. What motivated the namespace change then?

**Comment by davidfowl** [comment]
I think this might break Kestrel.

**Comment by KrzysztofCwalina** [comment]
@davidfowl, how  does it break Kestrel. Kestrel has a dependency on Piplelines.Extensions dll.

**Comment by davidfowl** [comment]
Kestrel doesn't use extensions though I guess it could in tests

**Comment by ahsonkhan** [comment]
@KrzysztofCwalina, what are we doing with this PR?

**Comment by ahsonkhan** [comment]
Closing this PR. Please feel free to re-open if necessary.

**Pull request closed without merging**